src/PrecompileTools.jl

```diff
@@ -1,27 +1,18 @@
 module PrecompileTools
 
-if VERSION >= v"1.6"
-    using Preferences
-end
+using Preferences
+
 export @setup_workload, @compile_workload, @recompile_invalidations
 
 const verbose = Ref(false)    # if true, prints all the precompiles
-const have_inference_tracking = isdefined(Core.Compiler, :__set_measure_typeinf)
-const have_force_compile = isdefined(Base, :Experimental) && isdefined(Base.Experimental, Symbol("#@force_compile"))
 
-function precompile_mi(mi)
+function precompile_mi(mi::Core.MethodInstance)
     precompile(mi.specTypes) # TODO: Julia should allow one to pass `mi` directly (would handle `invoke` properly)
```

Comment:
While here, might as well note that in v1.12, Julia does allow this
```suggestion
    precompile(mi)
```

(it also "allowed" this in previous versions, since JuliaLang/julia#46010, but those methods may segfault due to a typo (JuliaLang/julia#56595), so it is best to avoid calling that)

Reply:
Thanks, very helpful.